### PR TITLE
Clarify that eval_metric has moved from fit() to Predictor() (#1483)

### PR DESCRIPTION
**Pull request by willsmithorg**
*Issue #, if available:*
1483

*Description of changes:*
Clarify that eval_metric has moved from fit() to Predictor() 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1486-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1486/1/index.html

File: docs/tutorials/tabular_prediction/tabular-quickstart.md [hunk 1]
```diff
@@ -161,7 +160,7 @@ predictor_age = TabularPredictor(label=age_column, path="agModels-predictAge").f
...
+Note that we didn't need to tell AutoGluon this is a regression problem, it automatically inferred this from the data and reported the appropriate performance metric (RMSE by default). To specify a particular evaluation metric other than the default, set the `eval_metric` argument of `TabularPredictor()` and AutoGluon will tailor its models to optimize your metric (e.g. `eval_metric = 'mean_absolute_error'`). For evaluation metrics where higher values are worse (like RMSE), AutoGluon may sometimes flip their sign and print them as negative values during training (as it internally assumes higher values are better).
```
Comment (Innixma) [inline docs/tutorials/tabular_prediction/tabular-quickstart.md:163]: set the `eval_metric` argument of -> set the `eval_metric` parameter of

AutoGluon may sometimes flip their sign -> AutoGluon will flip their sign

File: examples/tabular/example_advanced_tabular.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 """ Example script for predicting columns of tables, demonstrating more advanced usage of fit().
     Note that all settings demonstrated here are just chosen for demonstration purposes (to minimize runtime), and do not represent wise choices to use in practice.
-    To maximize predictive accuracy, we recommend you do NOT specify `hyperparameters` or `hyperparameter_tune_kwargs`, and instead only specify the following fit() arguments: eval_metric=YOUR_METRIC, presets='best_quality'
+    To maximize predictive accuracy, we recommend you do NOT specify `hyperparameters` or `hyperparameter_tune_kwargs`, and instead only specify the arguments presets='best_quality' to fit() and eval_metric=YOUR_METRIC to TabularPredictor()
```
Comment (Innixma) [inline examples/tabular/example_advanced_tabular.py:3]: and instead specify `TabularPredictor(..., eval_metric=YOUR_METRIC).fit(..., presets='best_quality')`

**Review by Innixma — COMMENTED** [review]
Looks great! I added a couple comments to further clarify things.

File: core/src/autogluon/core/trainer/abstract_trainer.py [hunk 3]
```diff
@@ -60,7 +60,7 @@ def __init__(self, path: str, problem_type: str, eval_metric=None,
         if not (self.eval_metric.needs_pred or self.eval_metric.needs_quantile):
             logger.log(25, "\tThis metric expects predicted probabilities rather than predicted class labels, so you'll need to use predict_proba() instead of predict()")
 
-        logger.log(20, "\tTo change this, specify the eval_metric argument of fit()")
+        logger.log(20, "\tTo change this, specify the eval_metric argument of Predictor()")
```
[the remark this answers is not in the document]
Comment (willsmithorg) [reply core/src/autogluon/core/trainer/abstract_trainer.py]: Done

File: docs/tutorials/tabular_prediction/tabular-quickstart.md [hunk 4]
```diff
@@ -67,8 +67,7 @@ from autogluon.tabular import TabularPredictor
 predictor = TabularPredictor(label=<variable-name>).fit(train_data=<file-name>)
 ```
 
-**Note:** This simple call to `fit()` is intended for your first prototype model. In a subsequent section, we'll demonstrate how to maximize predictive performance by additionally specifying two `fit()` arguments: `presets` and `eval_metric`.
-
+**Note:** This simple call to `fit()` is intended for your first prototype model. In a subsequent section, we'll demonstrate how to maximize predictive performance by additionally specifying the `presets` argument to `fit()` and the `eval_metric` argument to `TabularPredictor()`.
```
[the remark this answers is not in the document]
Comment (willsmithorg) [reply docs/tutorials/tabular_prediction/tabular-quickstart.md]: Done

File: docs/tutorials/tabular_prediction/tabular-quickstart.md [hunk 5]
```diff
@@ -134,7 +133,7 @@ This command implements the following strategy to maximize accuracy:
...
 
-- Provide the `eval_metric` if you know what metric will be used to evaluate predictions in your application. Some other non-default metrics you might use include things like: `'f1'` (for binary classification), `'roc_auc'` (for binary classification), `'log_loss'` (for classification), `'mean_absolute_error'` (for regression), `'median_absolute_error'` (for regression).  You can also define your own custom metric function, see examples in the folder: `autogluon/core/metrics/`
+- Provide the `eval_metric` to `TabularPredictor()` if you know what metric will be used to evaluate predictions in your application. Some other non-default metrics you might use include things like: `'f1'` (for binary classification), `'roc_auc'` (for binary classification), `'log_loss'` (for classification), `'mean_absolute_error'` (for regression), `'median_absolute_error'` (for regression).  You can also define your own custom metric function, see examples in the folder: `autogluon/core/metrics/`
```
[the remark this answers is not in the document]
Comment (willsmithorg) [reply docs/tutorials/tabular_prediction/tabular-quickstart.md]: Done

File: docs/tutorials/tabular_prediction/tabular-quickstart.md (hunk 1 above)
Comment (willsmithorg) [reply docs/tutorials/tabular_prediction/tabular-quickstart.md]: Done

File: examples/tabular/example_advanced_tabular.py (hunk 2 above)
Comment (willsmithorg) [reply examples/tabular/example_advanced_tabular.py]: Done

**Comment by szha** [comment]
Job PR-1486-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1486/2/index.html

**Review by Innixma — APPROVED** [review]
LGTM!

**Pull request merged**